messaging/src/test/java/org/axonframework/eventhandling/replay/ReplayContextParameterResolverFactoryTest.java

```diff
@@ -0,0 +1,105 @@
+import static org.axonframework.eventhandling.GenericEventMessage.asEventMessage;
+import static org.junit.jupiter.api.Assertions.*;
+
+class ReplayContextParameterResolverFactoryTest {
+
+    private final MyResetContext resetContext = new MyResetContext(asList(2L, 3L));
+    private SomeHandler handler;
+    private AnnotationEventHandlerAdapter testSubject;
+    private ReplayToken replayToken;
+    private GlobalSequenceTrackingToken regularToken;
+
+    @BeforeEach
+    void setUp() {
+        handler = new SomeHandler();
+        testSubject = new AnnotationEventHandlerAdapter(handler);
+        regularToken = new GlobalSequenceTrackingToken(1L);
+        replayToken = (ReplayToken) ReplayToken.createReplayToken(regularToken,
+                                                                  new GlobalSequenceTrackingToken(0),
+                                                                  resetContext);
+    }
+
+    @Test
+    void testResolvesIfMatching() throws Exception {
+        GenericTrackedEventMessage<Object> replayEvent = new GenericTrackedEventMessage<>(replayToken,
+                                                                                          asEventMessage(1L));
+        GenericTrackedEventMessage<Object> liveEvent = new GenericTrackedEventMessage<>(regularToken,
+                                                                                        asEventMessage(2L));
+        assertTrue(testSubject.canHandle(replayEvent));
+        assertTrue(testSubject.canHandle(liveEvent));
+        testSubject.handle(createMessage(1L, true));
+        testSubject.handle(createMessage(2L, true));
+        testSubject.handle(createMessage(3L, true));
+        testSubject.handle(createMessage(4L, false));
+
+        assertEquals(asList(1L, 2L, 3L, 4L), handler.receivedLongs);
+        assertEquals(asList(2L, 3L), handler.receivedInReplay);
+    }
+
+    private TrackedEventMessage<Object> createMessage(Long position, boolean replay) {
+        return new GenericTrackedEventMessage<>(createToken(position, replay), asEventMessage(position));
+    }
+
+    private TrackingToken createToken(Long position, boolean replay) {
+        if (replay) {
+            return ReplayToken.createReplayToken(new GlobalSequenceTrackingToken(position + 1),
+                                                 new GlobalSequenceTrackingToken(position),
+                                                 resetContext);
+        }
+        return new GlobalSequenceTrackingToken(position);
+    }
+
+    private static class SomeHandler {
+
+        private List<Long> receivedLongs = new ArrayList<>();
+        private List<Long> receivedInReplay = new ArrayList<>();
+
+        @EventHandler
+        public void handle(Long event, TrackingToken token, @ReplayContext MyResetContext resetContext) {
+            receivedLongs.add(event);
+            if (resetContext != null && resetContext.sequences.contains(token.position().orElse(0))) {
```

Comment:
Curious: why have this `sequences.contains` check here deciding whether to add the `event` to the `receivedInReplay`? 
Isn't the fact the `resetContext` is not `null` not enough?

Reply:
I wanted to check whether it passes the right object into the token. Because of this check, event 1 is filtered out (it's not in the sequences) and we check this. 
If we only did a non-null check, the MyResetContext could be anything. It's something that would not quickly be wrongly implemented, but it also demonstrates how it could be used